### ISSUE TITLE
Terraform provider recognizing unexpected changes

### ISSUE THREAD

**Issue by pimguilherme** [gravitational/teleport-plugins#390]
Hi! 

It looks like the terraform provider got a bit lost when we tried reapplying the same TF code right after a successful apply. It detected some unexpected changes:
![image](https://user-images.githubusercontent.com/1406885/145580531-9f964f5f-ad5e-4d6f-8b4a-8964c13b9d40.png)

This is the code example we are using:
```tf
resource "teleport_role" "stone-devops" {

  metadata {
    name = "stone-devops"
  }

  spec {

    allow {

      app_labels {
        key   = "*"
        value = ["*"]
      }

      db_labels {
        key   = "*"
        value = ["*"]
      }

      db_names          = ["{{internal.db_names}}"]
      db_users          = ["{{internal.db_users}}"]
      kubernetes_groups = ["system:masters", "{{internal.kubernetes_groups}}"]

      kubernetes_labels {
        key   = "*"
        value = ["*"]
      }

      kubernetes_users = ["{{internal.kubernetes_users}}"]
      logins           = ["root", "{{internal.logins}}"]

      node_labels {
        key   = "*"
        value = ["*"]
      }

      rules {
        resources = ["trusted_cluster", "token", "lock"]
        verbs     = ["list", "create", "read", "update", "delete"]
      }

      rules {
        resources = ["session", "event", "role", "user"]
        verbs     = ["list", "read"]
      }
    }

    options {
      cert_format        = "standard"
      enhanced_recording = ["command", "network"]
      forward_agent      = false
      max_session_ttl    = "30h0m0s"
      port_forwarding    = true
    }
  }
}

```

This is the generated role:
```yaml
kind: role
metadata:
  id: 1639076796525113530
  name: stone-devops
spec:
  allow:
    app_labels:
      '*': '*'
    db_labels:
      '*': '*'
    db_names:
    - '{{internal.db_names}}'
    db_users:
    - '{{internal.db_users}}'
    kubernetes_groups:
    - system:masters
    - '{{internal.kubernetes_groups}}'
    kubernetes_labels:
      '*': '*'
    kubernetes_users:
    - '{{internal.kubernetes_users}}'
    logins:
    - root
    - '{{internal.logins}}'
    node_labels:
      '*': '*'
    rules:
    - resources:
      - trusted_cluster
      - token
      - lock
      verbs:
      - list
      - create
      - read
      - update
      - delete
    - resources:
      - session
      - event
      - role
      - user
      verbs:
      - list
      - read
  deny: {}
  options:
    cert_format: standard
    enhanced_recording:
    - command
    - network
    forward_agent: false
    max_session_ttl: 30h0m0s
    port_forwarding: true
version: v4
```

Would you please help us understand this issue?

Thanks!

**Comment by pimguilherme** [comment]
this seems to be resolved in version 8.0.7

**Comment by programmerq** [comment]
It looks like this commit is likely the specific commit that fixed the issue: https://github.com/gravitational/teleport-plugins/commit/2c10ffc2636aa10737bf122d8d6e355d6630ccae#diff-94dc3ea57721d541aae09b7bf2368c1e20d4c89996ff6df4349d86048877c0e7

It tells terraform that labels are added by teleport so terraform no longer has this behavior.
